mhlib/app.go

```diff
@@ -269,3 +293,81 @@ func (a *App) render(configFile string) (*string, *string, *[]byte, error) {
 
 	return &chart, &chartVersion, &overrides, nil
 }
+
+
+func selfRender(templateValuesStr string, enableGomplate bool) (string, error) {
+	/*
+	This function will accept an input string and run it through the
+	  templating engine as both the values dictionary as well as the
+	  template string.  It will repeat the process until the templating
+	  result stops changing.
+	*/
+
+	type Values map[string]interface{}
+
+	lastRender := templateValuesStr
+	for i := 0; i < 10; i++ {
+
+		// Unmarshal the file as a values dict
+		vals := Values{}
+		err := yaml.Unmarshal([]byte(templateValuesStr), &vals)
+		if err != nil { return "", err }
+
+		tmpl := template.New("SelfTemplate")
+		tmpl.Delims("[[", "]]")
+
+		if enableGomplate {
+			// Read the defined datasources and datasourcehaders
+			//   from the values dict
+			dataSources := []string{}
+			dataSourceHeaders := []string{}
+			if _, ok := vals["gomplate"]; ok {
```

Comment:
don't discard the value if you are going to look it up again on the next line yeah?

Reply:
Got it.  Made obsolete by next commit.